version: '2.1'

networks:
  chainpoint-node:
    driver: bridge

services:

  postgres:
    image: postgres:9.6.5-alpine
    container_name: postgres-node
    restart: always
    networks:
      - chainpoint-node
    volumes:
      - ./.data/postgresql:/var/lib/postgresql/data
    environment:
      POSTGRES_USER: chainpoint
      POSTGRES_PASSWORD: chainpoint

  redis:
    image: redis:4.0.1-alpine
    container_name: redis-node
    restart: always
    networks:
      - chainpoint-node
    volumes:
      - ./.redis/redis.conf:/usr/local/etc/redis/redis.conf:ro
      - ./.data/redis:/data
    command:
      - redis-server
      - /usr/local/etc/redis/redis.conf

  chainpoint-node:
    image: quay.io/chainpoint/chainpoint-node:1603d8116f53813a93105602c3150b4a714eb08e
    container_name: chainpoint-node
    restart: on-failure
    ports:
      - "80:8080"
    networks:
      - chainpoint-node
    depends_on:
      - redis
      - postgres
    environment:
      NODE_TNT_ADDRESS: ${NODE_TNT_ADDRESS}
      CHAINPOINT_NODE_PUBLIC_URI: "${CHAINPOINT_NODE_PUBLIC_URI:-http://0.0.0.0}"
